application_test.go

```diff
@@ -427,3 +427,20 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
+
+func TestWaitOnApplication(t *testing.T) {
+	endpoint := newFakeMarathonEndpoint(t, nil)
+	defer endpoint.Close()
+	client := endpoint.Client.(*marathonClient)
+
+	c := make(chan bool)
+	err := client.waitOnApplication(fakeAppName, c)
+	assert.NoError(t, err)
+
+	c = make(chan bool)
+	go func() {
+		c <- true
+	}()
```

Comment:
yep @dmajere code looks functionally identical, but far easier to read ... 

> The only real open question to me is whether we need the atomicSwitch for some reason. > @gambol99, you created the code in the first place. Can you shed some light on whether > @dmajere's implementation is equivalent to the previous code from your perspective?

As for my poor atomSwitch implementation, my only defense is go-marathon was my first stab at writing go :-)